app/src/main/java/com/vitorpamplona/amethyst/LocalPreferences.kt

```diff
@@ -14,104 +14,109 @@ import nostr.postr.events.Event.Companion.getRefinedEvent
 import nostr.postr.toHex
 
 class LocalPreferences(context: Context) {
-  val encryptedPreferences = EncryptedStorage().preferences(context)
-  val gson = GsonBuilder().create()
+    private object PrefKeys {
+        const val NOSTR_PRIVKEY = "nostr_privkey"
+        const val NOSTR_PUBKEY = "nostr_pubkey"
+        const val FOLLOWING_CHANNELS = "following_channels"
+        const val HIDDEN_USERS = "hidden_users"
+        const val RELAYS = "relays"
+        const val DONT_TRANSLATE_FROM = "dontTranslateFrom"
+        const val LANGUAGE_PREFS = "languagePreferences"
+        const val TRANSLATE_TO = "translateTo"
+        const val ZAP_AMOUNTS = "zapAmounts"
+        const val LATEST_CONTACT_LIST = "latestContactList"
+        val LAST_READ: (String) -> String = { route -> "last_read_route_${route}" }
+    }
 
-  fun clearEncryptedStorage() {
-    encryptedPreferences.edit().apply {
-      remove("nostr_privkey")
-      remove("nostr_pubkey")
-      remove("following_channels")
-      remove("hidden_users")
-      remove("relays")
-      remove("dontTranslateFrom")
-      remove("languagePreferences")
-      remove("translateTo")
-      remove("zapAmounts")
-      remove("latestContactList")
-    }.apply()
-  }
+    private val encryptedPreferences = EncryptedStorage().preferences(context)
+    private val gson = GsonBuilder().create()
 
-  fun saveToEncryptedStorage(account: Account) {
-    encryptedPreferences.edit().apply {
-      account.loggedIn.privKey?.let { putString("nostr_privkey", it.toHex()) }
-      account.loggedIn.pubKey.let { putString("nostr_pubkey", it.toHex()) }
-      account.followingChannels.let { putStringSet("following_channels", it) }
-      account.hiddenUsers.let { putStringSet("hidden_users", it) }
-      account.localRelays.let { putString("relays", gson.toJson(it)) }
-      account.dontTranslateFrom.let { putStringSet("dontTranslateFrom", it) }
-      account.languagePreferences.let { putString("languagePreferences", gson.toJson(it)) }
-      account.translateTo.let { putString("translateTo", it) }
-      account.zapAmountChoices.let { putString("zapAmounts", gson.toJson(it)) }
-      account.backupContactList.let { putString("latestContactList", Event.gson.toJson(it)) }
-    }.apply()
-  }
+    fun clearEncryptedStorage() {
+        encryptedPreferences.edit().apply {
+            encryptedPreferences.all.keys.forEach { remove(it) }
+        }.apply()
+    }
```

Comment:
As part of the existing PR, I noticed that logging out and logging back in resulted in the notification dots staying hidden. Traced that to the `clearEncryptedStorage` method not properly clearing those keys. Rather than listing all the keys out, we can just remove all of them.

The following two commits f34dea8 and 75901d8 are just cleanup while I was touching the file.